src/lib/models/useDeleteModelMutation.ts

```diff
@@ -0,0 +1,73 @@
+import { useDataEngine, useAlert } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { useMutation } from 'react-query'
+import { Schema } from '../useLoadApp'
+
+type MutationFnArgs = {
+    id: string
+    displayName: string
+    messages?: string[]
+}
+
+export function useDeleteModelMutation(schema: Schema) {
+    const engine = useDataEngine()
+    const { show: showDeletionSuccess } = useAlert(
+        ({ displayName, modelType }) =>
+            i18n.t('Successfully deleted {{modelType}} "{{displayName}}"', {
+                displayName,
+                modelType,
+            }),
+        { success: true }
+    )
+
+    const { show: showDeletionFailure } = useAlert(
+        ({ displayName, modelType, messages }) => {
+            if (messages) {
+                return i18n.t(
+                    'Failed to delete {{modelType}} "{{displayName}}"! {{messages}}',
+                    { displayName, modelType, messages: messages.join('; ') }
+                )
+            }
+
+            return i18n.t('Failed to delete {{modelType}} "{{displayName}}"!', {
+                displayName,
+                modelType,
+            })
+        },
+        { success: false }
+    )
+
+    return useMutation({
+        mutationFn: (variables: MutationFnArgs) => {
+            const alertPayload = {
+                modelType: schema.singular,
+                displayName: variables.displayName,
+            }
+
+            return engine
+                .mutate({
+                    resource: schema.plural,
+                    id: variables.id,
+                    type: 'delete',
+                })
+                .then((result) => {
```

Comment:
Why not use react query 's canonical way to handle this with eg. OnError and onSucess? This can potentially cause issues of you close the dialog in the middle of the request - and react query handles that for you.

Reply:
I've changed this so that error/success handling is now done in the `<DeleteAction/>` component, making this hook error- & success-handling agnostic. The `<DeleteAction/>` component now shows an alert on success while displaying a notice box in the modal on failure